packages/base-controller/CHANGELOG.md

```diff
@@ -6,6 +6,24 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [4.0.0]
+### Added
+- Add `ControllerGetStateAction` and `ControllerStateChangeEvent` types ([#1890](https://github.com/MetaMask/core/pull/1890), [#2029](https://github.com/MetaMask/core/pull/2029))
+- Add `NamespacedName` type ([#1890](https://github.com/MetaMask/core/pull/1890))
+  - This is the narrowest supertype of all names defined within a given namespace.
+
+### Changed
+- **BREAKING:** Narrow controller messenger `ActionConstraint['handler']` type to remove usage of `any` ([#1890](https://github.com/MetaMask/core/pull/1890))
+  - This type is now defined as the universal supertype of all action handlers, meaning any functions can be safely assigned to it, regardless of argument types, number of arguments, or return value type.
```

Comment:
```suggestion
- **BREAKING:** Alter controller messenger `ActionConstraint['handler']` type to remove usage of `any` ([#1890](https://github.com/MetaMask/core/pull/1890))
  - This type is now defined as the universal supertype of all functions, meaning any functions can be safely assigned as an action handler, regardless of argument types, number of arguments, or return value type.
```

Reply:
Okay. I wanted to make sure this wasn't a problem in case we were wrong, but I guess we've checked this multiple times in various ways, so it's probably safe.